docs/vectorizer/document-vectorization.md

```diff
@@ -0,0 +1,525 @@
+
+#### Minimal Document Table
+
+A minimal document source table requires only an identifier and a URI pointing to the document, this can be the same column:
+
+```sql
+CREATE TABLE documents (
+    uri TEXT PRIMARY KEY
+);
+
+-- Example records
+INSERT INTO documents (uri) VALUES 
+    ('s3://my-bucket/documents/product-manual.pdf'),
+    ('s3://my-bucket/documents/api-reference.md'),
+```
+
+#### Extended Document Table
+
+For your system, you can include any additional metadata that you might need to filter or classify documents. To make synchronization easier, you can also include a `created_at` and `updated_at` timestamp, which will trigger re-embedding when they are updated:
+
+```sql
+CREATE TABLE documents (
+    id SERIAL PRIMARY KEY,
+    title TEXT NOT NULL,
+    uri TEXT NOT NULL,
+    content_type TEXT,
+    created_at TIMESTAMPTZ DEFAULT CURRENT_TIMESTAMP,
+    updated_at TIMESTAMPTZ DEFAULT CURRENT_TIMESTAMP,
+    owner_id INTEGER,
+    access_level TEXT,
+    tags TEXT[]
+);
+
+-- Example with rich metadata
+INSERT INTO documents (title, uri, content_type, owner_id, access_level, tags) VALUES
+    ('Product Manual', 's3://my-bucket/documents/product-manual.pdf', 'application/pdf', 12, 'internal', ARRAY['product', 'reference']),
+    ('API Reference', 's3://my-bucket/documents/api-reference.md', 'text/markdown', 8, 'public', ARRAY['api', 'developer']);
+```
+
+#### Storing Document Content Directly
+
+For smaller documents or systems without external storage, you can also store content directly as binary data:
+
+```sql
+CREATE TABLE documents (
+    id SERIAL PRIMARY KEY,
+    file BYTEA,
+);
+```
+
+## Configuring Document Vectorizers
+
+A vectorizer is a declarative configuration that defines how documents are processed, chunked, and embedded. PGAI's vectorizer system automatically keeps document embeddings in sync with source documents. You can find the reference for vectorizers in the [API Reference documentation](./api-reference.md).
+
+### Basic Vectorizer Configuration
+
+Here's a complete vectorizer configuration for documents stored in S3:
+
+```sql
+SELECT ai.create_vectorizer(
+    'documents'::regclass,
+    loading => ai.loading_uri(column_name => 'uri'),
+    parsing => ai.parsing_auto(),
+    chunking => ai.chunking_recursive_character_text_splitter(
+        chunk_size => 700,
+        separators => array[E'\n## ', E'\n### ', E'\n#### ', E'\n- ', E'\n1. ', E'\n\n', E'\n', '.', '?', '!', ' ', '', '|']
+    ),
+    embedding => ai.embedding_openai('text-embedding-3-small', 768)     
+);
+```
+
+This configuration:
+1. Uses the `documents` table as the source
+2. Loads documents from URIs in the `uri` column
+3. Automatically detects and parses document formats
+4. Splits text into chunks at common markdown breaking points (headers, paragraphs, etc.)
+5. Generates embeddings using OpenAI's `text-embedding-3-small` model
+
+### Vectorizer Components
```

Comment:
Would it make sense this be located on top of the "basic vectorizer configuration" so the last one becomes more like the "config reference"?